corehq/apps/app_manager/decorators.py

```diff
@@ -146,5 +150,38 @@ def _no_conflict(*args, **kwargs):
     return _no_conflict
 
 
+def avoid_parallel_build_request(fn):
+    @wraps(fn)
+    def new_fn(app, comment, user_id, *args, **kwargs):
+        domain = app.domain
+        app_id = app.get_id
+        if _build_request_in_progress(domain, app_id):
+            raise BuildConflictException()
+        _set_build_in_progress_lock(domain, app_id)
+        fn_return = fn(app, comment, user_id, *args, **kwargs)
+        _release_build_in_progress_lock(domain, app_id)
+        return fn_return
+    return new_fn
+
+
+def _set_build_in_progress_lock(domain, app_id):
+    key = _build_request_cache_key(domain, app_id)
+    cache.set(key, True)
```

Comment:
What would you think of setting a timeout on this?

Reply:
how about an hour?

Reply:
sounds good to me!

Reply:
Done